src/components/modals/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { ButtonWithModal } from './ButtonWithModal';
+import { ActionButton } from './ModalWithActionButtons';
+
+export interface ButtonConfirmationModalProps {
+  modalTitle: string;
+  modalContentText: string;
+  name: string;
+  confirmButtonProps: ActionButton;
+  cancelButtonProps: ActionButton;
+}
+
+export const ButtonWithConfirmationModal = ({
+  modalTitle,
+  modalContentText,
+  name,
+  confirmButtonProps,
+  cancelButtonProps,
+  ...otherOpenButtonProps
+}: ButtonConfirmationModalProps) => {
+  return (
+    <ButtonWithModal
+      modalTitle={modalTitle}
+      modalContentText={modalContentText}
+      name={name}
+      actionButtonList={[cancelButtonProps, confirmButtonProps]}
+      {...otherOpenButtonProps}
+    />
+  );
+};
+
+export interface ButtonAlertModalProps {
+  modalTitle: string;
```

Comment:
again we should use modalProps instead of splattering title and contentText anywhere.

this is like the third time i've seen modalTitle/modalContentText in this pr, which means that in the future if we want to add a field to modalProps we have to add that in 3 places :))